### PR TITLE
Adding `util` python module and CI-Script

### LINKED ISSUE #106: Add CI
This assumes that our standard workflow is always creating PRs also from bots and lock the `master` (main) branch against modifications.

What we could check for:

- [x] Doublettes being added (check for doubles from fields like `liquipedia`, `Platforms->DE` and `Platforms->Voobly` ids, `name`, `aoeelo`)
  - [x] basically index those fields and throw an error if values already exists in the file
- [x] item IDs not being reused, kept unique (done in #107)
- [x] Probably implicit, but also valid YAML (done in #109)
- [x] Valid `ISO 3166-1 alpha-2` country code
  - [x] https://pypi.org/project/pycountry/
  - [x] https://pypi.org/project/iso3166/
- [ ] other stuff (additions?)

### PR DESCRIPTION
**Pull request by simonsan**
Fixes #106

Tasks:
- [x] Doublettes being added
  - [x] `id` 
  - [x] `liquipedia`
  - [x] `Platforms->DE` and `Platforms->Voobly` ids
  - [x] `name`
  - [x] `aoeelo`
  - [x] `esportsearnings`
- [x] item IDs not being reused, kept unique, actually having a `key` named `id`
- [x] Probably implicit, but also valid YAML (done in #109)
- [x] Valid `ISO 3166-1 alpha-2` country code

### CODE REVIEW

File: scripts/util/error.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    pass
+
+
+class ProcessingError(Exception):
+    """ Base class for errors in the processing stage """
+    pass
+
+
+class LintError(Exception):
+    """ Raised when an doublette was found """
```
Comment (simonsan) [inline scripts/util/error.py:12]:
```suggestion
    """  Base class for errors in the linting stage """
```

File: scripts/util/players.py [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+    errors = []
+    id_list = []
+    indexed_keys = ["id", "name", "aoeelo", "esportsearnings", "liquipedia",
+                    "country"]
+    # Don't throw an exeption for these keys if they are missing
+    voluntary_keys = ["aoeelo", "esportsearnings", "liquipedia", "country"]
+
+    # We don't check for duplicates here
+    non_unique_keys = ["country"]
```
Comment (simonsan) [inline scripts/util/players.py:46]: Refactor those three to a tuple, e.g. `("id", False, False)` -> `(key, voluntary, non-unique)`

File: scripts/util/requests.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+    url = 'https://liquipedia.net/ageofempires/api.php'
+
+    user_agent = 'https://github.com/SiegeEngineers/aoc-reference-data'
+
+    def __init__(self, debug=None, cache=None, game="aoe2"):
+
+        if debug is True:
+            self.debug = True
+            self.wait_secs = 3
```
Comment (happyleavesaoc) [inline scripts/util/requests.py:33]: technically we must wait 30 secs when using `askargs`

File: scripts/util/teams.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    id_list = []
+
+    # Tuple description
+    # (field, unique, optional, sub-key-settings)
+    # field: indexed keys
+    # unique: don't check for duplicates here
+    # optional: Don't throw an exeption for these keys if they are missing
+    # sub-key-settings: settings for contained sub-keys
+    index_key_settings = [
+        ('id', True, False, None),
```
Comment (happyleavesaoc) [inline scripts/util/teams.py:33]: Try a `namedtuple`!

File: scripts/util/data_processor.py [hunk 5]
```diff
@@ -0,0 +1,232 @@
+import logging
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+class DataProcessor(object):
+
+    def __init__(self):
```
Comment (happyleavesaoc) [inline scripts/util/data_processor.py:9]: Maybe `ci` should be an arg here instead of in multiple member functions?

**Review by happyleavesaoc — COMMENTED** [review]
Your Python is quite good! I left a few specific comments.

I am seeing some extra newlines in many files, not sure what that is.

**Pull request merged**